src/main/java/net/sf/jabref/logic/importer/ParserResult.java

```diff
@@ -46,36 +42,48 @@ public ParserResult(BibDatabase database) {
         this(database, new MetaData(), new HashMap<>());
     }
 
-    public ParserResult(BibDatabase base, MetaData metaData, Map<String, EntryType> entryTypes) {
-        this.base = base;
-        this.metaData = metaData;
-        this.entryTypes = entryTypes;
-        if (Objects.nonNull(base) && Objects.nonNull(metaData)) {
-            this.bibDatabaseContext = new BibDatabaseContext(base, metaData, file);
-        }
+    public ParserResult(BibDatabase database, MetaData metaData, Map<String, EntryType> entryTypes) {
+        this.database = Objects.requireNonNull(database);
+        this.metaData = Objects.requireNonNull(metaData);
+        this.entryTypes = Objects.requireNonNull(entryTypes);
     }
 
     public static ParserResult fromErrorMessage(String message) {
         ParserResult parserResult = new ParserResult();
         parserResult.addWarning(message);
+        parserResult.setInvalid(true);
         return parserResult;
     }
 
+    private static String getErrorMessage(Exception exception) {
+        String errorMessage = exception.getLocalizedMessage();
+        if (exception.getCause() != null) {
+            errorMessage += " Caused by: " + exception.getCause().getLocalizedMessage();
+        }
+        return errorMessage;
+    }
+
+    public static ParserResult getNullResult() {
```

Comment:
This smells a lot like singleton. There are potential problems in the current implementation, which do probably not manifest, but let us improve this anyway: If you always hand out the same `NULL_RESULT`, a caller of the method can modify this result (e.g. `NULL_RESULT.getDatabase().add(...)`), which could theoretically lead to interesting observations at other places.

Long story short: Please always create a new empty ParserResult here. Thus, you can also remove `NullResult`.